### PR TITLE
Enable TF 2.0 Object Detection API models

### PR DESCRIPTION
**Pull request by lazarevevgeny**
Description: Enabled models from https://github.com/tensorflow/models/blob/master/research/object_detection/g3doc/tf2_detection_zoo.md. Refer to the table below which models are enabled. The rest models are not planned to be enabled for now.

JIRA: 42435, 42005, 35524

Code:
* [x]  Comments
* [x]  Code style (PEP8)
* [x]  Transformation generates reshape-able IR
* [x]  Transformation preserves original framework node names

Validation:
* [x]  Unit tests: added unit tests where possible. The rest of graph transformations will be covered with the e2e tests
* [x]  Framework operation tests: N/A no new op enabled
* [x]  Transformation tests: added unit tests where possible. The rest of graph transformations will be covered with the e2e tests
* [x]  e2e model test with an update of ./tests/e2e_oss/utils/reshape_utils.py: 45284
* [x]  Model Optimizer IR Reader check: Manually checked

Documentation:
* [x]  Supported frameworks operations list: N/A no new op enabled
* [x]  Supported **public** models list: models have not reached supported status
* [x]  New operations specification: N/A no new operation enabled
* [x]  Guide on how to convert the **public** model
* [x]  User guide update

The following models are available and implemented in this PR:
 - [ ] CenterNet HourGlass104 512x512
 - [ ] CenterNet HourGlass104 Keypoints 512x512
 - [ ] CenterNet HourGlass104 1024x1024
 - [ ] CenterNet HourGlass104 Keypoints 1024x1024
 - [ ] CenterNet Resnet50 V1 FPN 512x512
 - [ ] CenterNet Resnet50 V1 FPN Keypoints 512x512
 - [ ] CenterNet Resnet101 V1 FPN 512x512
 - [ ] CenterNet Resnet50 V2 512x512
 - [ ] CenterNet Resnet50 V2 Keypoints 512x512
 - [x] EfficientDet D0 512x512                                 
 - [x] EfficientDet D1 640x640                                
 - [x] EfficientDet D2 768x768             
 - [x]  EfficientDet D3 896x896       
 - [x] EfficientDet D4 1024x1024            
 - [x] EfficientDet D5 1280x1280              
 - [x] EfficientDet D6 1280x1280                        
 - [x] EfficientDet D7 1536x1536              
 - [x] SSD MobileNet v2 320x320                       
 - [x] SSD MobileNet V1 FPN 640x640             
 - [x] SSD MobileNet V2 FPNLite 320x320        
 - [x] SSD MobileNet V2 FPNLite 640x640      
 - [x] SSD ResNet50 V1 FPN 640x640
 - [x] SSD ResNet50 V1 FPN 1024x1024
 - [x] SSD ResNet101 V1 FPN 640x640
 - [x] SSD ResNet101 V1 FPN 1024x1024
 - [x] SSD ResNet152 V1 FPN 640x640
 - [x] SSD ResNet152 V1 FPN 1024x1024
 - [x] Faster R-CNN ResNet50 V1 640x640               
 - [x] Faster R-CNN ResNet50 V1 1024x1024            
 - [x] Faster R-CNN ResNet50 V1 800x1333       
 - [x] Faster R-CNN ResNet101 V1 640x640           
 - [x] Faster R-CNN ResNet101 V1 1024x1024        
 - [x] Faster R-CNN ResNet101 V1 800x1333        
 - [x] Faster R-CNN ResNet152 V1 640x640              
 - [x] Faster R-CNN ResNet152 V1 1024x1024
 - [x] Faster R-CNN ResNet152 V1 800x1333        
 - [x] Faster R-CNN Inception ResNet V2 640x640      
 - [x] Faster R-CNN Inception ResNet V2 1024x1024           
 - [x] Mask R-CNN Inception ResNet V2 1024x1024
 - [ ] ExtremeNet

### CODE REVIEW

**Review by achetver — APPROVED** [review]
Looks ok to me.

**Review by jane-intel — COMMENTED** [review]
Is it true, that old config files work the same with updated Model Optimizer?

Also, don't we have a test for `sub_graph_between_nodes` function?

And one more question. Why we call such models as "trained **manually** using the TensorFlow\*". Why manually?

**Comment by lazarevevgeny** [comment]
> Is it true, that old config files work the same with updated Model Optimizer?
Yes, it is true. All changes are absolutely backward compatible. So even if customer has its own version of the config the model will still be converted successfully after my changes.

> Also, don't we have a test for `sub_graph_between_nodes` function?
Yes, there are tests. But I have not added tests for my change, my bad. Will add it.

> And one more question. Why we call such models as "trained **manually** using the TensorFlow*". Why manually?
Hmmm... I think that it was added because TF model zoo provides files trained with old version of the TF OD API, so these configs are used for models trained manually. But the word looks a bit confusing. Will remove it.

**Review by jane-intel — APPROVED** [review]

**Review by evolosen — APPROVED** [review]
